omod/src/main/webapp/resources/scripts/angular-common.js

```diff
@@ -1,32 +1,48 @@
-angular.module('uicommons.common', []).
-
-    factory('http-auth-interceptor', function($q, $rootScope) {
-        return {
-            responseError: function(response) {
-                if (response.status === 401 || response.status === 403) {
-                    $rootScope.$broadcast('event:auth-loginRequired');
-                }
-                return $q.reject(response);
-            }
-        }
-    }).
-
-    config(function($httpProvider) {
-        $httpProvider.interceptors.push('http-auth-interceptor');
-
-        // to prevent the browser from displaying a password pop-up in case of an authentication error
-        $httpProvider.defaults.headers.common['Disable-WWW-Authenticate'] = 'true';
-        
-        // prevent IE from caching GET request
-        $httpProvider.defaults.headers.common["Cache-Control"] = "no-cache";
-        $httpProvider.defaults.headers.common.Pragma = "no-cache";
-        $httpProvider.defaults.headers.common["If-Modified-Since"] = "0";
-    }).
-
-    run(['$rootScope', '$window', function ($rootScope, $window) {
-        $rootScope.$on('event:auth-loginRequired', function () {
-            if (confirm(emr.message("uicommons.notLoggedIn", "The operation cannot be completed, because you are no longer logged in. Do you want to go to login page?"))) {
-                window.location = "/" + OPENMRS_CONTEXT_PATH + "/login.htm";
-            }
-        });
+angular.module('uicommons.common', []).

```

Comment:
Sorry i was supposed to update the ticket description, this interceptor should be in a separate file add module, so that if a dev wishes to opt out it,  they just exclude the js file from their page